src/components/interactive-map/shapes/convoy-arrow.tsx

```diff
@@ -0,0 +1,116 @@
+type ConvoyArrowProps = {
+  x1: number;
+  y1: number;
+  x2: number;
+  y2: number;
+  x3: number;
+  y3: number;
+  lineWidth: number;
+  arrowWidth: number;
+  arrowLength: number;
+  offset: number;
+  fill: string;
+  stroke: string;
+  strokeWidth: number;
+  dash?: { length: number; spacing: number };
+  onRenderCenter?: (x: number, y: number, angle: number) => React.ReactElement;
+};
+
+const ConvoyArrow = (props: ConvoyArrowProps) => {
```

Comment:
I have a question mark in my head about this component.

I think the `shapes` should all be primitive shapes and not have game logic built in.

I actually think this component could just be a `Line` component. We can get the point to which the line goes in a separate helper function:
``` 
const { endX, endY } = getClosestPointOnLine(x1, y1, x2, y2, x3, y3)
return (
  <Line x1={x1} y1={y1} x2={endX} y2={endY} />
)
```

Reply:
This makes sense, but my plan is to use different lines. Right now the convoy line is straight, but i still want it to be wavey in the future. I'm okay with making primitives, although it's just one schema. The risk is combining different primitives will make it a bit harder to understand what is going on. I might take a look at this tomorrow if I have time